### PR TITLE
Feature/quizzes delete with AngularJS

### PR DESCRIPTION
**Pull request by anatoliiibrahimov**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/browse/LR-183)

### Author.

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @sergiy17 
- [x] @brouhaha88 
- [x] @vovka

### Issue summary.

Reworked delete with AngularJS

### Fix summary. 

Reworked delete with AngularJS

### Testing approach.

Added test for delete Quiz.

### CODE REVIEW

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 1]
```diff
@@ -98,6 +110,14 @@ describe("indexQuizCtrl", function(){
       expect($scope.reverse).toEqual(reverseState);
     });
   });
+   
+  describe('$scope.deleteQuiz', function() {
+    it('delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
+      $scope.deleteQuiz();
```
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: defined by AngularJS

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 2]
```diff
@@ -72,10 +72,22 @@ describe("newQuizCtrl", function() {
...
 
   beforeEach(inject(function(_$controller_){
     $controller = _$controller_;
+
+    $controller('indexQuizCtrl', {
+      $scope: $scope,
+      Quiz: Quiz
+    });
+
+    $scope.quiz = {};
```
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: defined by AngularJS

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 3]
```diff
@@ -72,10 +72,22 @@ describe("newQuizCtrl", function() {
...
+  $scope = {};
+  Quiz = {
+    delete: function() {},
+  };
 
   beforeEach(inject(function(_$controller_){
     $controller = _$controller_;
+
+    $controller('indexQuizCtrl', {
+      $scope: $scope,
```
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: defined by AngularJS

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 4]
```diff
@@ -72,10 +72,22 @@ describe("newQuizCtrl", function() {
 
 describe("indexQuizCtrl", function(){
   beforeEach(module("videoQuiz"));
-  var $controller;
+  var $controller, Quiz;
+
+  $scope = {};
```
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: defined by AngularJS

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 5]
```diff
@@ -98,6 +110,14 @@ describe("indexQuizCtrl", function(){
       expect($scope.reverse).toEqual(reverseState);
     });
   });
+   
+  describe('$scope.deleteQuiz', function() {
+    it('delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
```
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: defined by AngularJS

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 6]
```diff
@@ -70,26 +70,38 @@ describe("newQuizCtrl", function() {
...
 });
 
-describe("indexQuizCtrl", function(){
-  beforeEach(module("videoQuiz"));
-  var $controller;
+describe('indexQuizCtrl', function(){
+  beforeEach(module('videoQuiz'));
+  var $controller, Quiz;
+
+  $scope = {};
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Add `var`

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 3 above)
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: It's not. You defined it above.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 2 above)
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: You defined it by yourself above.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 5 above)
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: It's defined by jasmine, not angular.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 1 above)
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: You defined it by yourself above.

**Comment by vovka** [comment]
1. Please, separate test phases with 1 empty line.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 5 above)
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it was automatic comment. Cuz it always says that something is not defined

**Comment by anatoliiibrahimov** [comment]
@vovka 
I changed all quotes cuz  everytime bot says mixed quotes. If we need double qoutes i will change it.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 7]
```diff
@@ -70,26 +70,38 @@ describe("newQuizCtrl", function() {
...
+  describe('$scope.sort', function(){
+    it('should change the sorting key', function(){
       var $scope = {};
-      $controller("indexQuizCtrl", {$scope: $scope});
+      $controller('indexQuizCtrl', {$scope: $scope});
 
-      $scope.sort("name");
-      expect($scope.sortKey).toEqual("name");
+      $scope.sort('name');
+      expect($scope.sortKey).toEqual('name');
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["expect" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2425317163/issues/source?bid=3205144&fileBranchId=3239403#l99)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 8]
```diff
@@ -70,26 +70,38 @@ describe("newQuizCtrl", function() {
...
+      Quiz: Quiz
+    });
+
+    $scope.quiz = {};
   }));
 
-  describe("$scope.sort", function(){
-    it("should change the sorting key", function(){
+  describe('$scope.sort', function(){
+    it('should change the sorting key', function(){
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["it" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2425317163/issues/source?bid=3205144&fileBranchId=3239403#l94)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 6 above)
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["$scope" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2425317163/issues/source?bid=3205144&fileBranchId=3239403#l77)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 9]
```diff
@@ -70,26 +70,38 @@ describe("newQuizCtrl", function() {
...
+      $controller('indexQuizCtrl', {$scope: $scope});
 
-      $scope.sort("name");
-      expect($scope.sortKey).toEqual("name");
+      $scope.sort('name');
+      expect($scope.sortKey).toEqual('name');
     });
 
-    it("should change the order of sorting quizzes to reverse", function(){
+    it('should change the order of sorting quizzes to reverse', function(){
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["it" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2425317163/issues/source?bid=3205144&fileBranchId=3239403#l102)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 10]
```diff
@@ -70,26 +70,38 @@ describe("newQuizCtrl", function() {
...
-describe("indexQuizCtrl", function(){
-  beforeEach(module("videoQuiz"));
-  var $controller;
+describe('indexQuizCtrl', function(){
+  beforeEach(module('videoQuiz'));
+  var $controller, Quiz;
+
+  $scope = {};
+  Quiz = {
+    delete: function() {},
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: [Expected method shorthand.](https://www.codacy.com/app/vovka/video_quiz/file/2425317163/issues/source?bid=3205144&fileBranchId=3239403#l79)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 11]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+      $scope.confirm = function(msg) { return true; };
+
+      $scope.deleteQuiz();
+ 
+      expect(Quiz.delete).toHaveBeenCalled();
+    });
+  });
 
-        $scope.add();
+   describe('$scope.deleteQuiz', function() {
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: This `decsribe` is redundant. Group two delete test inside one describe.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 12]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+      $controller('indexQuizCtrl', {$scope: $scope});
+
+      var reverseState = true;
+
+      $scope.sort();
+
+      expect($scope.reverse).toEqual(reverseState);
+    });
+  });
+
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Are all those changes above yours? ^^^

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 13]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
 
-        $scope.add();
+   describe('$scope.deleteQuiz', function() {
+    it('cancel delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
+      $scope.confirm = function(msg) { return false; };
 
-        expect($scope.quiz.questions_attributes.length).toEqual(2);
-      });
+      expect($scope.confirm).not.toBe(true);
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: How do you think:
1. what equals `$scope.confirm` here?
2. Can the value of `$scope.confirm` be changed?

Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: we can change it to (Quiz.delete).not.toBe(true) . Test won't fall to

Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Could you please answer two questions?

Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: 1.It equals false.
2.Yes we can, but in other case our test will have no reason.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 12 above)
Comment (brouhaha88) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: all our branches go from Nazars test commit from indexQuiz, thats the reason he has this code same as everybody else

**Comment by brouhaha88** [comment]
Marked as checked, but please DO take into account Vova's comment on your test

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 14]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+      $scope.sort();
+
+      expect($scope.reverse).toEqual(reverseState);
+    });
+  });
+   
+  describe('$scope.deleteQuiz', function() {
+    it('delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
+      $scope.confirm = function(msg) { return true; };
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ['msg' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l117)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 15]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+ 
+      expect(Quiz.delete).toHaveBeenCalled();
+    });
+  });
 
-        $scope.add();
+   describe('$scope.deleteQuiz', function() {
+    it('cancel delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
+      $scope.confirm = function(msg) { return false; };
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ['msg' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l128)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 16]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
-  });
+    $scope.quiz = {};
+  }));
 
-  describe("newQuizCtrl", function() {
-    describe("$scope.add", function() {
-      it("Add new input to the form", function() {
-        var $scope = {};
-        $controller("newQuizCtrl", {$scope: $scope});
+  describe('$scope.sort', function(){
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["describe" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l93)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 17]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+
+    it('should change the order of sorting quizzes to reverse', function(){
+      var $scope = {};
+      $controller('indexQuizCtrl', {$scope: $scope});
+
+      var reverseState = true;
+
+      $scope.sort();
+
+      expect($scope.reverse).toEqual(reverseState);
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["expect" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l110)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 18]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
-      it("Add new input to the form", function() {
-        var $scope = {};
-        $controller("newQuizCtrl", {$scope: $scope});
+  describe('$scope.sort', function(){
+    it('should change the sorting key', function(){
+      var $scope = {};
+      $controller('indexQuizCtrl', {$scope: $scope});
+
+      $scope.sort('name');
+      expect($scope.sortKey).toEqual('name');
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["expect" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l99)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 19]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
-        $scope.sort("name");
+      expect($scope.quiz.questions_attributes.length).toEqual(2);
+    });
+  });
+});
 
-        expect($scope.sortKey).toEqual("name");
-      });
+describe('indexQuizCtrl', function(){
+  beforeEach(module('videoQuiz'));
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["beforeEach" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l74)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 20]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+    $scope.quiz = {};
+  }));
 
-  describe("newQuizCtrl", function() {
-    describe("$scope.add", function() {
-      it("Add new input to the form", function() {
-        var $scope = {};
-        $controller("newQuizCtrl", {$scope: $scope});
+  describe('$scope.sort', function(){
+    it('should change the sorting key', function(){
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["it" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l94)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 21]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+      var $scope = {};
+      $controller('indexQuizCtrl', {$scope: $scope});
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: [Expected property shorthand.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l96)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 22]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+describe('indexQuizCtrl', function(){
+  beforeEach(module('videoQuiz'));
+  var $controller, $scope, Quiz;
 
-      it("should change the order of sorting quizzes to reverse", function(){
-        var $scope = {};
-        $controller("indexQuizCtrl", {$scope: $scope});
-        $scope.reverse = false;
+  $scope = {};
+   Quiz = {
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l78)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 23]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+   describe('$scope.deleteQuiz', function() {
+    it('cancel delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
+      $scope.confirm = function(msg) { return false; };
 
-        expect($scope.quiz.questions_attributes.length).toEqual(2);
-      });
+      expect($scope.confirm).not.toBe(true);
     });
   });
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 0.](https://www.codacy.com/app/vovka/video_quiz/file/2440330913/issues/source?bid=3205144&fileBranchId=3239403#l132)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 16 above)
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it's just as it should be

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 18 above)
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it's just as it should be

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 17 above)
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it's just as it should be

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 19 above)
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it's just as it should be

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 20 above)
Comment (anatoliiibrahimov) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it's just as it should be

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 24]
```diff
@@ -52,47 +52,80 @@ describe("Bind Cocoon events to link", function(){
...
+      expect(Quiz.delete).toHaveBeenCalled();
+    });
 
-        expect($scope.quiz.questions_attributes.length).toEqual(2);
-      });
+    it('cancel delete quiz from quizzes list', function() {
+      spyOn(Quiz, 'delete');
+      $scope.confirm = function(msg) { return false; };
+ 
+      expect($scope.confirm).not.toBe(true);
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: what equals $scope.confirm here?
Can the value of $scope.confirm be changed?

**Comment by vovka** [comment]
Please, make this PR to reflect only your changes.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 25]
```diff
@@ -52,47 +52,82 @@ describe("Bind Cocoon events to link", function(){
...
+
+      var reverseState = true;
+
+      $scope.sort();
+
+      expect($scope.reverse).toEqual(reverseState);
+    });
   });
+   
+  describe('$scope.dele teQuiz', function() {
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: A typo here. Please, fix it.

**Comment by anatoliiibrahimov** [comment]
@vovka done

**Comment by vovka** [comment]
OK, looks good

**Pull request merged**